raven/gis.py

```diff
@@ -0,0 +1,472 @@
+from collections import Counter
+from osgeo import gdal, gdal_array, ogr, osr
+from warnings import warn
+import logging
+import numpy as np
+import os
+
+logging.captureWarnings(True)
+gdal.UseExceptions()
+
+
+# TODO: Integrate some of the functionality from rasterio and/or shapely and/or fiona
+class RavenGIS(object):
+    """
+    Base object for GIS analyses. Contains static methods that are used to perform data extractions of raster
+    and vector data objects as well as performs comparison analyses between them. Ideally, common functions for both
+    such as CRS functions, extent, area calculations and reprojections will be refactored here.
+    """
+
+    def __init__(self):
+        self._shape = None
+        self._shape_reproj = None
+        self._raster = None
+        self._raster_reproj = None
+        self._pretty_crs = None
+        self._epsg = None
+
+    def shape(self):
+        return self._shape
+
+    def raster(self):
+        return self._raster
+
+    def pretty_crs(self):
+        return self._pretty_crs
+
+    def crs(self):
+        return self._epsg
+
+    @staticmethod
+    def raster_bbox_corners(raster):
+        """Returns the corner coordinates from a raster grid"""
+        upper_left_x, x_res, x_skew, upper_left_y, y_skew, y_res = raster.GetGeoTransform()
+
+        lower_right_x = upper_left_x + (raster.RasterXSize * x_res)
+        lower_right_y = upper_left_y + (raster.RasterYSize * y_res)
+        return upper_left_x, upper_left_y, lower_right_x, lower_right_y
+
+    @staticmethod
```

Comment:
xd, yd are the width and height, correct ? I think this should be documented. It's not just a coordinate transformation.